frameworks/AutoGluon/exec.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import os
+import warnings
+warnings.simplefilter("ignore")
+
+import matplotlib
+import numpy as np
+import pandas as pd
+matplotlib.use('agg')  # no need for tk
+
+from autogluon.task.tabular_prediction.tabular_prediction import TabularPrediction as task
+from autogluon.utils.tabular.utils.savers import save_pd, save_pkl
+import autogluon.utils.tabular.metrics as metrics
+
+from frameworks.shared.callee import call_run, result, Timer, touch
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info("\n**** AutoGluon ****\n")
+
+    metrics_mapping = dict(
+        acc=metrics.accuracy,
+        auc=metrics.roc_auc,
+        f1=metrics.f1,
+        logloss=metrics.log_loss,
+        mae=metrics.mean_absolute_error,
+        mse=metrics.mean_squared_error,
+        r2=metrics.r2
+    )
+
+    perf_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if perf_metric is None:
+        # TODO: figure out if we are going to blindly pass metrics through, or if we use a strict mapping
+        log.warning("Performance metric %s not supported.", config.metric)
+
+    is_classification = config.type == 'classification'
+
+    train = pd.DataFrame(dataset.train.data, columns=dataset.columns)
+    label = dataset.target.name
+
+    output_dir = make_subdir("models", config)
+    with Timer() as training:
+        predictor = task.fit(
```

Comment:
We should pass problem_type here as some problems are ambiguous between being regression and multiclass.